batches: Fix YAML reference docs subtitle hyperlinks

st0nebraker opened the pull request:
Closes #51197 - Fixes YAML reference docs subtitle hyperlinks. The extra markdown linking wasn't necessary & causing interference it seems to break the links. `#` permalink displays now.

## Test plan
- Nav to `docs.sourcegraph.com/batch_changes/references/batch_spec_yaml_reference` locally and ensure table of contents & subtitle hyperlinks work as expected

courier-new reviewed (approved):
Great!! I wonder if this was something that changed when we updated the markdown renderer for the docsite. Glad these headers "just work" now without the added annotation.

courier-new commented:
Can you tell, is this broken in all releases of 5.0? I'd love to plan to backport this to Patch #4 in a couple weeks if so.

st0nebraker commented:
> Great!! I wonder if this was something that changed when we updated the markdown renderer for the docsite. Glad these headers "just work" now without the added annotation.

Do you know when we updated the .md renderer?

> Can you tell, is this broken in all releases of 5.0? I'd love to plan to backport this to Patch 4 in a couple weeks if so.

I believe it is, I just added the backport label to get in there!

courier-new commented:
> Do you know when we updated the .md renderer?
I do not, and it's not obvious to me from recent commits either. I just remember it coming up in Slack semi-recently...

Do you mind dropping the `please-backport-5.0` label on this instead? It looks like that's the one [the release guild is asking us to use](https://sourcegraph.slack.com/archives/C0EPTDE9L/p1683915374570279) for the 5.0 patches.
